Posix compat

ShellyHerself opened the pull request:
Likely not complete yet. Mainly for Mozzarilla Purposes.

The filepicker library needs to be tested yet, which is separate from this.

ShellyHerself committed:
Fix terrible default linux color picker

ShellyHerself committed:
Merge remote-tracking branch 'remotes/origin/develop' into posix-compat

ShellyHerself committed:
Merge remote-tracking branch 'origin/develop' into posix-compat

ShellyHerself committed:
Disable blocks start hidden for linux to avoid problems with certain window managers.

ShellyHerself committed:
Fix tag window string.

ShellyHerself committed:
Fix config saving breaking with posix paths.

ShellyHerself committed:
Fix handler for posix

ShellyHerself committed:
Fixed a few paths not being converted to strings before the config gets serialized.

ShellyHerself committed:
Remove a debug print.

ShellyHerself committed:
Merge remote-tracking branch 'develop' into posix-compat

ShellyHerself committed:
Switch util.path references to supyr_struct

ShellyHerself committed:
Fix my utter stupidity.

ShellyHerself committed:
[app_window] Fix except pass-ing keyboard interrupts

ShellyHerself committed:
[filedialog] Use wrapper for filedialog so we have a better one on Linux

ShellyHerself committed:
[filedialog] Temporary fix for import order issue

ShellyHerself committed:
[hek/handler] Case insensitive my ass

ShellyHerself commented on __init__.py:
```diff
@@ -13,3 +13,9 @@
+# conflict to do with FieldWidget.
+# TODO: Moses, please fix the root of this issue.
+from binilla.windows import filedialog
+del filedialog
```
Any way we can fix this @MosesofEgypt?

MosesofEgypt replied:
Will try to look into this today/tonight
EDIT: Fixed. Will push changes tonight.

ShellyHerself commented on defs/config_def.py:
```diff
@@ -106,7 +107,10 @@
+    DEFAULT=(
+        sum([1<<i for i in (2, 3, 4, 6, 7, 8)])
+        if e_c.IS_LNX else
+        sum([1<<i for i in (2, 3, 4, 6, 7, 8, 9, 10)]))
```
A comment explaining this would be nice.

MosesofEgypt replied:
Done. Will be included in update tonight.

ShellyHerself commented on widgets/field_widgets/color_picker_frame.py:
```diff
@@ -1,7 +1,7 @@
 import tkinter.ttk as ttk
 
-from tkinter.colorchooser import askcolor
+from tkcolorpicker import askcolor
```
Add `tkcolorpicker` to the dependencies.

ShellyHerself reviewed (commented):
Some personal notes.

MosesofEgypt committed:
Tweaked various uses of pathlib.Path objects so they work below python 3.6(os.path functions don't accept Path objects below 3.6)
Changing all uses of filepath strings to use pathlib.Path objects instead.
Removed unused methods and properties from Handler.

MosesofEgypt committed:
Made several Binilla properties decorated to always be a Path object.
Moved logfile creation into apply_config so it can be created if not done at startup.
Fixed log never being created.
Moved logfile into location of config file.

MosesofEgypt committed:
Made it so an error won't be reported when load_tags is called on a tag that is loaded, but doesn't have a tag window associated yet.
Created extra method to get the tag window id of a given tag.

ShellyHerself committed:
[app_window] Safe config location for Linux

MosesofEgypt committed:
Fixed bug in ComputedTextFrame where it would not update text if frame was ever disabled.

MosesofEgypt committed:
Fixed bug with circular dependency due to filedialog being in binilla.windows, and field_widgets importing it, but tag_window requiring field_widgets
Moved assignment of ColorPickerFrame.has_alpha into reload and populate methods, as the value may need to be updated after initialization of the widget
Added comment to setting of default flags in config def
Added tkcolorpicker to setup.py requirements

MosesofEgypt committed:
Added property to allow configuring the config window class instance created when opening config
Split TagWindow.__init__ into two halves: pre and post toplevel init. Post is its own method, allowing for subclassing.
Added code to allow tag windows to have their iconbitmap customized

ShellyHerself committed:
[tag_window] Simplify icon bitmap setting. Report on failure.

ShellyHerself committed:
[app_window] Fix not assiging default config path on Linux

MosesofEgypt committed:
Better fix for circular dependency bug in binilla.windows(also fixes issue for refinery)

MosesofEgypt committed:
Modified get_cwd to return a more sane working directory

MosesofEgypt committed:
Finished removing reliance on PATHDIV

MosesofEgypt committed:
Removed reliance on sanitize_path

ShellyHerself committed:
[defs] Improve comment

ShellyHerself committed:
[util] Remove get_cwd in favor of more portable methods

ShellyHerself committed:
Merge pull request #7 from MosesofEgypt/py_35_compat

Py 35 compat
